packages/contracts-bridge/contracts/test/BridgeRouter.t.sol

```diff
@@ -297,6 +716,266 @@ contract BridgeRouterTest is BridgeTest {
         assertEq(revertingToHook.test(), 123);
     }
 
+    function testFuzz_handleHookTransferSucceeds(
+        uint256 tokenAmount,
+        bytes32 tokenDetailsHash,
+        bytes memory extraData,
+        bytes32 sender,
+        uint32 nonce
+    ) public {
+        tokenAmount = bound(
+            tokenAmount,
+            0,
+            type(uint256).max - bridgeUserTokenAmount
+        );
+        bytes32 hook = address(revertingToHook).addressToBytes32();
+        // The hook will succed only if origin < 10. This has nothing
+        // to do with how transferToHook works, but rather how the mock works.
+        // We created a mock that will have different behaviour depending on the origin,
+        // so that we can test different scenarios easily.
+        uint32 origin = 1;
+        localToken.mint(address(bridgeRouter), tokenAmount);
+        bytes memory action = abi.encodePacked(
+            BridgeMessage.Types.TransferToHook,
+            hook,
+            tokenAmount,
+            tokenDetailsHash,
+            sender,
+            extraData
+        );
+        bytes memory tokenId = abi.encodePacked(
+            localDomain,
+            address(localToken).addressToBytes32()
+        );
+        // The hook succeeds
+        bridgeRouter.exposed_handleTransferToHook(
+            origin,
+            nonce,
+            tokenId,
+            action
+        );
+        assertEq(revertingToHook.test(), 123);
+    }
+
+    event Receive(
```

Comment:
should be at top of contract, or immediately above test that uses it

Reply:
will do the second